### PR TITLE
ENH: Poll for viewer before calling functions

### PR DESCRIPTION
**Pull request by bnmajor**
This PR starts a new background job in a separate thread when a Viewer method is called. If the Viewer does not yet exist it polls until it does before calling the method.

async_demo

### CODE REVIEW

File: itkwidgets/viewer.py [hunk 1]
```diff
@@ -144,115 +146,141 @@ def __init__(
...
             ui_collapsed=ui_collapsed, rotate=rotate, ui=ui, **add_data_kwargs
         )
+        self.queue = queue.Queue()
+        self.bg_thread = self.bg_jobs.new(self.queue_worker)
         api.export(self.viewer_rpc)
 
+    async def run_queued_requests(self):
+        while not hasattr(self.viewer_rpc, 'itk_viewer'):
+            await asyncio.sleep(1)
+        while hasattr(self.viewer_rpc, 'itk_viewer'):
```
[the remark this answers is not in the document]
Comment (bnmajor) [reply itkwidgets/viewer.py]: Fixed

File: itkwidgets/viewer.py [hunk 2]
```diff
@@ -144,115 +146,141 @@ def __init__(
...
+        task = loop.create_task(fn(*args))
+        loop.run_until_complete(task)
+
+    def queue_worker(self):
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        task = loop.create_task(self.run_queued_requests())
+        loop.run_until_complete(task)
+
+    def queue_request(self, method, *args):
```
[the remark this answers is not in the document]
Comment (bnmajor) [reply itkwidgets/viewer.py:184]: Fixed

File: itkwidgets/viewer.py [hunk 3]
```diff
@@ -144,115 +149,143 @@ def __init__(
...
         api.export(self.viewer_rpc)
 
+    @property
+    def loop(self):
+        return asyncio.get_running_loop()
+
+    async def run_queued_requests(self):
+        self.viewer_rpc.event.wait()
+        while self.queue.qsize():
+            await asyncio.sleep(1)
```
Comment (thewtex) [inline itkwidgets/viewer.py:167]: Replace with callback for setImage, etc., remove sleeps

Comment (bnmajor) [reply itkwidgets/viewer.py]: Updated, no longer sleeps. Instead uses events and queues to manage what methods are called when. All data setters and UI methods are queued to be called immediately after the viewer is created. Any data property setters are added to a deferred queue and called as soon as the data has been set.

**Review by thewtex — APPROVED** [review]

**Pull request merged**